test/metabase/api/embed_test.clj

```diff
@@ -47,28 +48,54 @@
                 :params   {}}
                additional-token-params)))
 
-(defmacro with-temp-card {:style/indent 1} [[card-binding & [card]] & body]
-  `(let [card-defaults# ~card
-         card-settings# (merge (when-not (:dataset_query card-defaults#)
-                                 (public-test/count-of-venues-card))
-                               card-defaults#)]
-     (mt/with-temp Card [~card-binding card-settings#]
-       ~@body)))
+(defn do-with-temp-card [m f]
+  (let [m (merge (when-not (:dataset_query m)
+                   {:dataset_query (mt/mbql-query venues {:aggregation [[:count]]})})
+                 m)]
+    (mt/with-temp Card [card m]
+      (f card))))
 
-(defmacro with-temp-dashcard {:style/indent 1} [[dashcard-binding {:keys [dash card dashcard]}] & body]
-  `(with-temp-card [card# ~card]
-     (mt/with-temp* [Dashboard     [dash# ~dash]
-                     DashboardCard [~dashcard-binding (merge {:card_id      (u/the-id card#)
-                                                              :dashboard_id (u/the-id dash#)}
-                                                             ~dashcard)]]
-       ~@body)))
+(defmacro with-temp-card {:style/indent 1} [[binding & [card]] & body]
+  `(do-with-temp-card
+    ~card
+    (fn [~binding]
+      ~@body)))
+
+(defn do-with-temp-dashcard [{:keys [dash card dashcard]} f]
+  (with-temp-card [card card]
+    (mt/with-temp* [Dashboard     [dashboard (merge
+                                              (when-not (:parameters dash)
+                                                {:parameters [{:id      "_VENUE_ID_"
+                                                               :name    "Venue ID"
+                                                               :slug    "venue_id"
+                                                               :type    "id"
+                                                               :target  [:dimension (mt/id :venues :id)]
+                                                               :default nil}]})
+                                              dash)]
+                    DashboardCard [dashcard  (merge {:dashboard_id       (u/the-id dashboard)
+                                                     :card_id            (u/the-id card)
+                                                     :parameter_mappings (or (:parameter_mappings dashcard)
+                                                                             [{:parameter_id "_VENUE_ID_"
+                                                                               :card_id      (u/the-id card)
+                                                                               :target       [:dimension [:field (mt/id :venues :id) nil]]}])}
+                                                    dashcard)]]
+      (f dashcard))))
```

Comment:
I had to rework this mock stuff here to create Dashboards that actually have real parameters. Otherwise the new unified param resolution code will complain that there's no matching parameter. That seems like a win anyway to me